/*
 * This file was automatically generated by EvoSuite
 * Fri Mar 02 22:23:57 GMT 2018
 */

package java_programs;

import org.junit.Test;
import static org.junit.Assert.*;
import java_programs.LIS;

public class LIS_ESTest {

	@Test(timeout = 4000)
	public void test0() throws Throwable {
		LIS lIS0 = new LIS();
	}

	@Test(timeout = 4000)
	public void test1() throws Throwable {
		int[] intArray0 = new int[7];
		intArray0[0] = 1317;
		int int0 = LIS.lis(intArray0);
		assertEquals(1, int0);
	}

	@Test(timeout = 4000)
	public void test2() throws Throwable {
		// Undeclared exception!
		try {
			LIS.lis((int[]) null);
			fail("Expecting exception: NullPointerException");

		} catch (NullPointerException e) {
			//
			// no message in exception (getMessage() returned null)
			//
		}
	}

	@Test(timeout = 4000)
	public void test3() throws Throwable {
		int[] intArray0 = new int[0];
		int int0 = LIS.lis(intArray0);
		assertEquals(0, int0);
	}

	@Test(timeout = 4000)
	public void test4() throws Throwable {
		int[] intArray0 = new int[7];
		intArray0[4] = 1317;
		intArray0[5] = 1317;
		int int0 = LIS.lis(intArray0);
		assertEquals(2, int0);
	}

	@Test(timeout = 4000)
	public void test5Failing() throws Throwable {
		fail("Fail");
	}
}
